A36-xds-for-servers.md

```diff
@@ -334,6 +339,44 @@ connection, it must still be checked when verifying that the most-specific
 matching logic is guaranteed to produce a unique result during Listener
 validation.
 
+### Resource Validation
+
+The `Listener` proto validation rules are modified as follows:
+
+ - The `listener_filters` field must be empty.
+ - The `use_original_dst` field must not be set with the value `true`.
+ - For every entry in the `filter_chains` field:
+   - The `filter_chain_match` field must be normalized as described in the
+     [FilterChainMatch section](#filterchainmatch) and there must not be
+     duplication between any normalized matchers in the `Listener`.
+ - For the `default_filter_chain` field and every entry in the `filter_chains`
+   field:
+   - The `filters` field must have exactly one entry. Inside of it:
+     - The `type_url` field must have the value
+       `envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager`
+     - Inside the `value` field, decoded as a `HttpConnectionManager` message:
+       - The `http_filters` field must be validated as specified in
+         [A39: xDS HTTP Filter Support][A39].
+       - If any `RouteConfiguration` features are supported:
+         - Either the `route_config` field or the `rds` field must be set
+         - If the `route_config` field is set, its value must be validated
+           according to the `RouteConfiguration` validation rules.
+
+> [!NOTE]
+> Once other `Fitler` types are supported, the validation logic will need to
+> be expanded to accept other `Filter` types. `Filter` types are the type contained in the
+> [`typed_config`][Filter.typed_config] Any. If the type is
+> `udpa.type.v1.TypedStruct`, then its [`type_url`][TypedStruct.type_url] is used
+> instead. The `HttpConnectionManager` filter must always be present in the
+> last entry in each `filters` list.
+
+The `RouteConfiguration` proto validation rules (predomenently defined in [gRFC A28][A28-validation]) are modified as follows:
+
+ - Every entry of `virtual_hosts` is validated, not just the most specifically matching one.
+ - For each entry of `virtual_hosts`:
+   - For each entry of `routes`:
+     - The `action` field can now have the value `non_forwarding_action` in addition to other accepted values.
```

Comment:
I think we should explicitly call out that this means that a parsed route can now have multiple types of actions: RouteAction (usable for clients), NonForwardingAction (usable for servers), or unknown (an action type not currently supported by gRPC).  On clients, any action type other than RouteAction will cause RPCs to fail with UNAVAILABLE; on servers, any action type other than NonForwardingAction will cause RPCs to fail with UNAVAILABLE.  And as per my comment above, we should also say something about what information should be included in the RPC failure status message.